### PR TITLE
Reward Model based Evaluations

### PR DESCRIPTION
**Pull request by sanderland**
This adds a section on reward model based evaluations. 

I've used these a lot, but not seen a lot written about them, possibly due to high-performing open reward models being a fairly recent development.
They also have a number of pitfalls that I've seen again and again (like the absolute scores being somewhat meaningless).

This probably needs a bit of work on references and backlinking, but thought I'd start here and we can iterate if you think this would be an interesting addition.

### CODE REVIEW

**Comment by clefourrier** [comment]
Hi! Yes, sounds super cool, I have not found a lot of literature on those!

I would organize it in the following way.
```
# Reward Model Based Evaluation
## What is a reward-model?
includes your intro, the bradley terry models, and the Other types of reward models section - since they are short I'm not sure they need sub-headers.
I'd love to see a bit more references there, notably if you know of cool blogs which could explain the concepts easily

## How do I use a reward-model for evaluation?

## Pros and cons of reward models

## Tips and tricks of using reward models for evaluation
```

**Comment by sanderland** [comment]
Have used your proposed setup and linked some example models, but trying to avoid the focus shifting to purely reward models themselves rather than evaluations.
Don't know a lot of cool blogs but if I tag @natolambert he might come along and link.

**Review by natolambert — APPROVED** [review]
The paper I know of that uses reward models most as a judge is Nemotron https://arxiv.org/abs/2406.11704.

Second, I'm aware of another type of eval done with reward models, which is to look at the average score from a very big reward model on a fixed set of eval prompts. Normally, I've heard, this score from a very strong reward model tends to be a bit smoother than looking at noisy eval scoring itself. It seems like it'll be a bit until we have a paper documenting this.

File: contents/Model as a judge/Reward Model.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+This model has the advantage of requiring only pairwise comparisons, which are easier to collect.
+However, as these models are trained only on comparisons, it is important to keep in mind that scores across different prompts should not be compared,
+ as they can be arbitrarily offset.
+
+Some reward models such as [SteerLM](https://arxiv.org/abs/2311.09528) output one or more absolute scores, which can be used to evaluate the model directly without the need for paired completions.
+These models can be easier to use for evaluation, but are also harder to collect data for. 
+More recently models have been proposed that output both absolute and relative scores, such as [HelpSteer2-Preference](https://arxiv.org/abs/2410.01257) and [ArmoRM](https://arxiv.org/abs/2406.12845).
+
+Another type of reward model are pair-wise models, which output a single score for a pair of completions ([example](https://huggingface.co/RLHFlow/pair-preference-model-LLaMA3-8B)).
```
[the remark this answers is not in the document]
Comment (sanderland) [reply contents/Model as a judge/Reward Model.md:20]: Thanks for the reference!

> the average score from a very big reward model on a fixed set of eval prompts

This is the main thing I tried to describe, with the caveat

* Average (BT) score, easy but can be affected by outlier scores and differences in confidence inherent in prompts, since scores are in logit space. 
  *  -> use win rate to avoid this, a little granular
     *  -> use mean win probability, smooth and not so affected by outliers.
   
Of course average absolute score doesn't have outliers, but the models tend to be a little weaker.
